Public/Src/Engine/Dll/EngineSchedule.cs

```diff
@@ -744,6 +744,12 @@ private bool ShouldSerializeOptimizationDataStructurePostExecution()
             IEnumerable<string> nonScrubbablePaths,
             ITempDirectoryCleaner tempCleaner)
         {
+            if (mountPathExpander != null)
+            {
+                // We don't scrub mounts explicitly marked as non-scrubbable
+                nonScrubbablePaths = nonScrubbablePaths.Concat(mountPathExpander.GetNonScrubbableRoots().Select(p => p.ToString(scheduler.Context.PathTable)));
+            }
+
             var pathsToScrub = new List<string>();
             if (configuration.Engine.Scrub && mountPathExpander != null)
             {

```

Comment:
I think the configuration for scrubbing has gotten pretty complicated. 
1. There are scrubbable mounts, where anything except declared build outputs get deleted. But this only happens if the /scrub flag is passed
2. /scrubdirectory specifies some directories should be scrubbed. This happens regardless of whether the specified directory is under a scrubbable mount or not
3. Shared opaque directories use the directory scrubber, but there "scrub" is a bit different. It really means delete the previous build's outputs

All of these obeyed nonScrubbablePaths, which were special paths one should never scrub, like the cache directory.

Adding all mounts that aren't marked scrubbable to the nonScrubbablePaths modified the existing behavior of the above, in particular I don't think we want #3